ledger/src/test_helpers/chain_builder.rs

```diff
@@ -181,32 +222,69 @@ impl<N: Network> TestChainBuilder<N> {
             last_block_round: 0,
             round_to_certificates: Default::default(),
             previous_leader_certificate: Default::default(),
+            transactions,
         })
     }
 
     /// Create multiple blocks, with fully-connected DAGs.
     pub fn generate_blocks(&mut self, num_blocks: usize, rng: &mut TestRng) -> Result<Vec<Block<N>>> {
-        self.generate_blocks_with_opts(num_blocks, GenerateBlocksOptions::default(), rng)
+        let num_validators = self.private_keys.len();
+        let transactions = mem::take(&mut self.transactions);
```

Comment:
Is this the only purpose of setting `transactions`? If so, wouldn't it be cleaner to not have that member and just call `generate_block_with_opts` from the testchain generator?

Reply:
we could - I just wanted to use the current setup as much as possible, in order to not mix things up; and as it stands now, _the_ pick of choice is `generate_blocks` (which is currently being batched "from the outside" - I think it would be preferable to move that batching into the `chain_builder` setup)